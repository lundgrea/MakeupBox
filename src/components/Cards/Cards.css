.card {
  border-radius: 5px;
  min-width: 30vw;  
  margin: 10px 2px 0px 2px;
  padding: 10px;
  text-align: center;
  background: white;
  color: #242B28;
}

.card-description {
  overflow: hidden;
  -webkit-line-clamp: 2;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}

.card-bran d{
  font-weight: 800;
  font-size: 32px;
  margin: 10px 0px 10px 0px;
}

.card-img {
  height: 200px;
  margin: 10px 0px 10px 0px
}

