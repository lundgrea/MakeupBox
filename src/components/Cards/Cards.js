import React from 'react'
import './Cards.css'


export const Cards = (props) => {
  return (
    <article className="card">
      <img className="card-img" alt="product" src={props.img}></img>
      <h3 className="card-brand">{props.brand}</h3>
      <h4 className="card-name">{props.name}</h4>
      <p className="card-description">{props.description}</p>
      <h6 className="card-price">${props.price}</h6>
      <p className="card-tags">{props.tags}</p>
      {/* <p>{props.color}</p> */}
    </article>
  )
}

export default Cards;