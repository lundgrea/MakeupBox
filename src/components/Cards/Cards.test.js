import React from 'react';
import { shallow } from 'enzyme';
import Cards from './Cards';

describe('Cards', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Cards 
        id={1}
        brand='Maybelline'
        type='Lipstick'
        description='The best lipstick around'
        img='url'
        name='Matte Life'
        price='7.00'
        tags={['gluten free']}
        key={1}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});