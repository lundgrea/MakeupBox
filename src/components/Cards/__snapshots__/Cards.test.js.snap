// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should match the snapshot 1`] = `
<article
  className="card"
>
  <img
    alt="product"
    className="card-img"
    src="url"
  />
  <h3
    className="card-brand"
  >
    Maybelline
  </h3>
  <h4
    className="card-name"
  >
    Matte Life
  </h4>
  <p
    className="card-description"
  >
    The best lipstick around
  </p>
  <h6
    className="card-price"
  >
    $
    7.00
  </h6>
  <p
    className="card-tags"
  >
    gluten free
  </p>
</article>
`;

exports[`Cards should match the snapshot with all data passed in correctly 1`] = `
<article
  className="card"
>
  <img
    alt="product"
    className="card-img"
    src="url"
  />
  <h3
    className="card-brand"
  >
    Maybelline
  </h3>
  <h4
    className="card-name"
  >
    Matte Life
  </h4>
  <p
    className="card-description"
  >
    The best lipstick around
  </p>
  <h6
    className="card-price"
  >
    $
    7.00
  </h6>
  <p
    className="card-tags"
  >
    gluten free
  </p>
</article>
`;
